### PR TITLE
📖 Document multiple providers support

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

Explicitly highlight in the documentation that CAPI supports running multiple providers on the same management cluster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/7234

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Looks good!

Only issue with adding this is that this isn't something that's currently tested in e2e tests, so while it should work, if it breaks there's no signal for it.

**Comment by fabriziopandini** [comment]
+1 to add a disclaimer that there are no E2E tests currently testing combinations of providers, so users willing to rely on such features are recommended to set up custom validation of CAPI before deploying on production

**Comment by alexander-demichev** [comment]
@killianmuldoon @fabriziopandini I added a warning about it

File: docs/book/src/tasks/multiple-providers.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Cluster API supports running multiple infrastructure/boostrap/control plane providers on the same management cluster. It's highly recommeded to rely on
+[clusterctl init](../clusterctl/commands/init.md) command in this case. [clusterctl](../clusterctl/overview.md) will help ensure that all providers support the same
+[API Version of Cluster API](../clusterctl/provider-contract.md#metadata-yaml) (contract).
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+The case of running multiple providers is not covered in Cluster API E2E suite. It's recommended to set up a manual validation of it
+before deploying providers on a production environment.
```
Comment (fabriziopandini) [inline docs/book/src/tasks/multiple-providers.md:12]: nit: What about
```suggestion
Currently, the case of running multiple providers is not covered in Cluster API E2E suite. It's recommended to set up a custom validation pipeline of
 your specific combination of providers before deploying it on a production environment.
```

Comment (alexander-demichev) [reply docs/book/src/tasks/multiple-providers.md]: 👍

File: docs/book/src/tasks/multiple-providers.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Running multiple providers
+
+Cluster API supports running multiple infrastructure/boostrap/control plane providers on the same management cluster. It's highly recommeded to rely on
```
Comment (dharmjit) [inline docs/book/src/tasks/multiple-providers.md:3]: nit: typo s/boostrap/bootstrap

File: docs/book/src/tasks/multiple-providers.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+[clusterctl init](../clusterctl/commands/init.md) command in this case. [clusterctl](../clusterctl/overview.md) will help ensure that all providers support the same
+[API Version of Cluster API](../clusterctl/provider-contract.md#metadata-yaml) (contract).
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+Currently, the case of running multiple providers is not covered in Cluster API E2E suite. It's recommended to set up a custom validation pipeline of
```
Comment (dharmjit) [inline docs/book/src/tasks/multiple-providers.md:11]: nit: s/API E2E suite/API E2E test suite

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Pull request merged**